### PR TITLE
Beta brede

### PR DESCRIPTION
**Pull request by BredeFK**
## Changed from `NTNU-Bachelor-Management-System-For-CS-Assignments` to `CSAMS` in project
* Also added so that new users have to confirm email before being created

## New/updated tables
```mysql
CREATE TABLE `validation`
(
    `id`        int(11)     NOT NULL AUTO_INCREMENT,
    `hash`      varchar(64) NOT NULL,
    `user_id`   int(11)              DEFAULT NULL,
    `valid`     tinyint(1)  NOT NULL DEFAULT 1,
    `timestamp` datetime    NOT NULL,
    PRIMARY KEY (`id`)
);

CREATE TABLE `users_pending`
(
    `id`            int(11)     NOT NULL AUTO_INCREMENT,
    `name`          varchar(64) DEFAULT NULL,
    `email_student` varchar(64) NOT NULL,
    `password`      varchar(64) NOT NULL,
    `validation_id` int(11)     NOT NULL,
    PRIMARY KEY (`id`),
    FOREIGN KEY (`validation_id`) REFERENCES validation (`id`)
);
```

### CODE REVIEW

File: webservice/controller/forgottenPass.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func changePasswordPOST(password string, hash string, w http.ResponseWriter, r *
 
 	// Services
 	//userService := service.NewUserService(db.GetDB())
-	forgottenService := service.NewForgottenPassService(db.GetDB())
+	forgottenService := service.NewValidationService(db.GetDB())
```
Comment (JohanAanesen) [inline webservice/controller/forgottenPass.go]: forgottenservice should be validationservice

File: webservice/model/user_pending.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package model
+
+// UserPending struct for keeping user_pending
+type UserPending struct {
```
Comment (kongebra) [inline webservice/model/user_pending.go]: More descriptive maybe?
Eg. UserRegistrationgPending

File: webservice/model/validation.go [hunk 3]
```diff
@@ -2,8 +2,8 @@ package model
 
 import "time"
 
-// ForgottenPass struct for keeping the data for forgotten password
-type ForgottenPass struct {
+// Validation struct for keeping the data for forgotten password
+type Validation struct {
```
Comment (kongebra) [inline webservice/model/validation.go]: More descriptive naming.

This is validation of what? Everything?

File: webservice/repositroy/validation.go [hunk 4]
```diff
@@ -2,34 +2,34 @@ package repositroy
...
-	"github.com/JohanAanesen/NTNU-Bachelor-Management-System-For-CS-Assignments/webservice/model"
-	"github.com/JohanAanesen/NTNU-Bachelor-Management-System-For-CS-Assignments/webservice/shared/util"
+	"github.com/JohanAanesen/CSAMS/webservice/model"
+	"github.com/JohanAanesen/CSAMS/webservice/shared/util"
 )
 
-// ForgottenPassRepository struct
-type ForgottenPassRepository struct {
+// ValidationRepository struct
+type ValidationRepository struct {
```
Comment (kongebra) [inline webservice/repositroy/validation.go]: Non-descriptive naming

File: webservice/repositroy/user_pending.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
+package repositroy
+
+import (
+	"database/sql"
+	"github.com/JohanAanesen/CSAMS/webservice/model"
+)
+
+// UserPendingRepository struct
+type UserPendingRepository struct {
```
Comment (kongebra) [inline webservice/repositroy/user_pending.go]: Non-descriptive naming

File: webservice/service/validation.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+import (
+	"database/sql"
+	"github.com/JohanAanesen/CSAMS/webservice/model"
+	"github.com/JohanAanesen/CSAMS/webservice/repositroy"
+	"github.com/JohanAanesen/CSAMS/webservice/shared/util"
+)
+
+// ValidationService struct
+type ValidationService struct {
```
Comment (kongebra) [inline webservice/service/validation.go]: Non-descriptive naming

File: docker-compose.yml [hunk 7]
```diff
@@ -74,7 +74,7 @@ services:
       PEER_AUTH: "5243980712315079823517089"
       MAIL_AUTH: "RqHDMR6WfMd9zMhr^kgDA4kZ0DJbEElu7Sb^B!#1uXVzDy%zW85yGIL#iI6D2%S*"
       USERNAME: cs.assignments.noreply@gmail.com
-      PASSWORD:
+      PASSWORD: YoqZ^92y#ab
```
[the remark this answers is not in the document]
Comment (JohanAanesen) [reply docker-compose.yml]: this is fixed, see discord

**Pull request closed without merging**